core/server/src/main/java/alluxio/master/DefaultAlluxioMaster.java

```diff
@@ -0,0 +1,411 @@
+import alluxio.web.MasterWebServer;
+import alluxio.web.WebServer;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+import org.apache.thrift.TMultiplexedProcessor;
+import org.apache.thrift.TProcessor;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.server.TServer;
+import org.apache.thrift.server.TThreadPoolServer;
+import org.apache.thrift.server.TThreadPoolServer.Args;
+import org.apache.thrift.transport.TServerSocket;
+import org.apache.thrift.transport.TTransportFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * This class encapsulates the different master services that are configured to run.
+ */
+@NotThreadSafe
+public class DefaultAlluxioMaster implements AlluxioMasterService {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  /** Maximum number of threads to serve the rpc server. */
+  private final int mMaxWorkerThreads;
+
+  /** Minimum number of threads to serve the rpc server. */
+  private final int mMinWorkerThreads;
+
+  /** The port for the RPC server. */
+  private final int mPort;
+
+  /** The socket for thrift rpc server. */
+  private final TServerSocket mTServerSocket;
+
+  /** The transport provider to create thrift server transport. */
+  private final TransportProvider mTransportProvider;
+
+  /** The address for the rpc server. */
+  private final InetSocketAddress mRpcAddress;
+
+  private final MetricsServlet mMetricsServlet = new MetricsServlet(MetricsSystem.METRIC_REGISTRY);
+
+  /** The master managing all block metadata. */
+  protected BlockMaster mBlockMaster;
+
+  /** The master managing all file system related metadata. */
+  protected FileSystemMaster mFileSystemMaster;
+
+  /** The master managing all lineage related metadata. */
+  protected LineageMaster mLineageMaster;
+
+  /** A list of extra masters to launch based on service loader. */
+  protected List<Master> mAdditionalMasters;
+
+  /** The journal for the block master. */
+  protected final ReadWriteJournal mBlockMasterJournal;
+
+  /** The journal for the file system master. */
+  protected final ReadWriteJournal mFileSystemMasterJournal;
+
+  /** The journal for the lineage master. */
+  protected final ReadWriteJournal mLineageMasterJournal;
+
+  /** The web ui server. */
+  private WebServer mWebServer = null;
+
+  /** The RPC server. */
+  private TServer mMasterServiceServer = null;
+
+  /** is true if the master is serving the RPC server. */
+  private boolean mIsServing = false;
+
+  /** The start time for when the master started serving the RPC server. */
+  private long mStartTimeMs = -1;
+
+  protected DefaultAlluxioMaster() {
```

Comment:
since `DefaultAlluxioWorker` constructor now throws exception, shall we make `DefaultAlluxioMaster` similar?